internal/rbd/controllerserver.go

```diff
@@ -175,6 +175,33 @@ func getGRPCErrorForCreateVolume(err error) error {
 	return status.Error(codes.Internal, err.Error())
 }
 
+// validateRequestedVolumeSize validates the request volume size with the
+// source snapshot or volume size, if there is a size missmatch it returns an error.
+func validateRequestedVolumeSize(rbdVol, parentVol *rbdVolume, rbdSnap *rbdSnapshot, cr *util.Credentials) error {
+	if rbdSnap != nil {
+		vol := generateVolFromSnap(rbdSnap)
+		err := vol.Connect(cr)
+		if err != nil {
```

Comment:
```suggestion
		if err := vol.Connect(cr); err != nil {
```

Reply:
Will it fail in correct model? Any ci check will fail?

Reply:
If I do short variable declared with if I need to se for other places also, I prefer this way as its used  in most of the places